### ISSUE TITLE
[BUG] Simple arithmetic error for real variables

### ISSUE THREAD

**Issue by schmittjoaopedro** [chocoteam/choco-solver#675]
## Expected behavior
Executing the following code with real numbers is inconsistent to constraints before implications. After define that `x = 1`, constraints that check the exact value `x = 1 -> y = 2` are not passing. However, if I use `x >= 0.999 -> y = 2` it works.

```java
Model model = new Model();
RealVar x = model.realVar(1.0, 3.0, 0.001);
RealVar y = model.realVar(1.0, 3.0, 0.001);
x.eq(1.0).ibex(0.001).post();
x.eq(1.0).ibex(0.001).reify().imp(y.eq(2.0).ibex(0.001).reify()).post(); // <- It doesn't works
//x.ge(1.0).ibex(0.001).reify().imp(y.eq(2.0).ibex(0.001).reify()).post(); // <- It doesn't works
//x.ge(0.999).ibex(0.001).reify().imp(y.eq(2.0).ibex(0.001).reify()).post(); // <- It works
model.getSolver().propagate();
System.out.println(x); // Should be [1.0 .. 1.0]
System.out.println(y); // Should be [2.0 .. 2.0]        
```

The following code was executed with ibex 2.8.6 and choco 4.10.2 using Java 1.8.0_231x86.

Please, could you fix this bug?

Thanks

**Comment by cprudhom** [comment]
We have started a new PR which proposes [a pure Java alternative to Ibex](#666 ).
You can work directly with the `continuous branch` if you want to give a try, while waiting for the branch to be merged into `master`.

The code would look like:
```java
Model model = new Model();
RealVar x = model.realVar(1.0, 3.0, 0.001);
RealVar y = model.realVar(1.0, 3.0, 0.001);
x.eq(1.0).equation().post();
x.eq(1.0).equation().reify().imp(y.eq(2.0).equation().reify()).post(); // <- It doesn't works
model.getSolver().propagate();
System.out.println(x); // Should be [1.0 .. 1.0]
System.out.println(y); // Should be [2.0 .. 2.0]
```
that is `.ibex(double)` is replaced by `.equation()`.

**Comment by schmittjoaopedro** [comment]
I started to test the equation API available in the 4.10.3-snapshot version. I suppose that it will provide the functionality that I need.

Thanks
